### PR TITLE
chore: CLOUDP-215162: Update Jira GitHub Action to update Ticket status based on the issue

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description
Ticket: [CLOUDP-215162](https://jira.mongodb.org/browse/CLOUDP-215162)

This PR updates the Jira issue action as follows:
-  The action will close the related Jira ticket if the GitHub issue is closed
- The action will reopen the related Jira ticket if the issue is closed

I tested my changes in a personal repo: 

- Issue: https://github.com/andreaangiolillo/cdk-test/issues/46
- Related Jira ticket: https://jira.mongodb.org/browse/CLOUDP-217786

## Next Steps:

Since we can test the logic only after the PR is merged, we will enable these changes only for TF. We will add these changes to other integrations once we are confident that the automation covers all the edge cases.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: .github/workflows/issues.yml [hunk 1]
```diff
@@ -3,13 +3,14 @@
...
     issues:
-      types: [opened]
+      types: [opened, reopened, closed]
   permissions: 
     issues: write
     contents: read
   jobs:
     jira_task:
       name: Create Jira issue
+      if: github.event_name == 'issues' && github.event.action == 'opened'
```
Comment (lantoli) [inline .github/workflows/issues.yml]: knitpit: do we need to check github.event_name == 'issues' ? 
it looks like it's the only "on" trigger

Comment (andreaangiolillo) [reply .github/workflows/issues.yml]: no good catch. I will remove it

**Review by lantoli — APPROVED** [review]

File: .github/workflows/issues.yml (hunk 1 above)
Comment (andreaangiolillo) [reply .github/workflows/issues.yml]: updated

**Review by lantoli — APPROVED** [review]

**Review by marcosuma — APPROVED** [review]
